libs/ui/src/components/activity-section-list/activity-section-list.tsx

```diff
@@ -45,16 +47,47 @@ const renderSectionHeader = (item: { section: SectionListData<ActivityData, Sect
 
 const emptyIconSize = getCustomSize(isWeb ? 30 : 36);
 
-export const ActivitySectionList: FC<Props> = ({ publicKeyHash, chainId, tokenAddress = '' }) => {
+export const ActivitySectionList: FC<Props> = ({ publicKeyHash, chainId, filterType, tokenAddress = '' }) => {
   const [offsetY, setOffsetY] = useState(0);
-  const { activity, fetch, isLoading } = useAllActivity(publicKeyHash, getDebankId(chainId), tokenAddress);
+  const { activity: allActivity, fetch, isLoading } = useAllActivity(publicKeyHash, getDebankId(chainId), tokenAddress);
+
+  const [activity, setActivity] = useState(allActivity);
 
   const handleFetchData = () => {
     if (offsetY === 0) {
       fetch(0);
     }
   };
 
+  const filterActivity = (condition: (activity: ActivityData) => boolean) => {
+    setActivity(
+      allActivity
+        .map(item => ({
+          title: item.title,
+          data: item.data.filter(activity => condition(activity))
+        }))
+        .filter(item => item.data.length)
```

Comment:
mb better to pass activity type, not a function ? 
and why we do map and filter after it, mb better to do just filter?

Reply:
transaction type does not match sorting, there is such a field as collections
because we need to filter only data items, and the filter is needed in order to remove empty titles without activity